File: euler_038.py
from time import time
from math import ceil

def is_pandigital(s):
    return "".join(sorted(s)) == "123456789"

def concat(st_num, n):
    res = ""
    for k in range(1, n + 1):
        res += str(st_num * k)
    return res

start = time()

c = set()

for n in range(2,10):
    for st in range(1, 10**ceil(9/n)):
        con = concat(st, n)
        if len(con) != 9:
            continue
        else:
            if is_pandigital(con):
                c.add((int(con), st, n))

print("Largest pandigital number: " + str(max([e[0] for e in c])))


stop = time()
print("Time: " + str(stop-start))

File: euler_038.js
const start = Date.now();

const isPandigital = (str) => {
    return str.split("").sort().join("") === "123456789";
}

const concatProduct = (num,n) => {
    let prod = "";
    for (let i = 1; i < n + 1; i++) {
        prod += (i * num).toString();
    }

    return prod;
}

let largestPN = 918273645;

for (let n = 2; n < 10; n++) {
    let ul = 10**(Math.ceil(9 / n));
    
    for (let num = 1; num <= ul; num++) {
        let cp = concatProduct(num,n);
        if (cp.length > 9) {
            break;
        }
        if (cp.length === 9 && Number(cp) > largestPN && isPandigital(cp)) {
            largestPN = Number(cp);
        }
    }
}


const end = Date.now();

console.log("Execute time: ", end - start, " Largest pandigital number is: ", largestPN);